src/lib.rs

```diff
@@ -309,6 +318,64 @@ impl PyIterProtocol for PyEmbeddings {
     }
 }
 
+/// finalfusion vocab.
+#[pyclass(name=Vocab)]
+struct PyVocab {
+    embeddings: Rc<RefCell<EmbeddingsWrap>>,
+}
+
+#[pymethods]
+impl PyVocab {
+    fn item_to_indices(&self, key: String) -> PyResult<PyObject> {
+        let embeds = self.embeddings.borrow();
+
+        use EmbeddingsWrap::*;
+        let indices = match &*embeds {
```

Comment:
I was thinking the same. Getting the VocabWrap could be implemented, and maybe some methods that are shared among the different Storage implementations.

It's not possible to implement Deref for EmbeddingsWrap because of the type parameters, right?

Reply:
> It's not possible to implement Deref for EmbeddingsWrap because of the type parameters, right?

Right, but this would go a long way in simplifying the API:

~~~rust
impl EmbeddingsWrap {
    fn storage(&self) -> &Storage {
        use EmbeddingsWrap::*;

        match self {
            View(e) => e.storage(),
            NonView(e) => e.storage(),
        }
    }

    fn vocab(&self) -> &VocabWrap {
        use EmbeddingsWrap::*;

        match self {
            View(e) => e.vocab(),
            NonView(e) => e.vocab(),
        }
    }

}
~~~